src/ucp/core/ucp_am.c

```diff
@@ -951,6 +957,13 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_am_send_nbx,
 
     UCP_WORKER_THREAD_CS_ENTER_CONDITIONAL(worker);
 
+    if (ENABLE_PARAMS_CHECK && ucs_unlikely(id > UINT16_MAX)) {
+        ret = UCS_STATUS_PTR(UCS_ERR_INVALID_PARAM);
+        ucs_error("Invalid AM id %u, must be in range [0, %u]",
```

Comment:
1. small case
2. try to unite common code with ucp_worker_set_am_recv_handler(), a helper function like ucp_am_check_id(unsigned am_id) that will print the error message and return failure status if am id is bad

Reply:
2. so we prefer to use ENABLE_PARAMS_CHECK condition on both cases? (fast **and slow** path)?

Reply:
@yosefe 
2. after adding a helper function
```
static ucs_status_t ucp_am_check_id(unsigned am_id)
{
    if (ENABLE_PARAMS_CHECK && ucs_unlikely(am_id > UINT16_MAX)) {
        printf("invalid AM id %u, must be in range [0, %u]",
                  am_id, UINT16_MAX);
        return UCS_ERR_INVALID_PARAM;
    }

    return UCS_OK;
}
```
I get a very strange warning that fails gtest and it seems to be unrelated (at least not directly related).
This is the output:

```
$ GTEST_FILTER=*rc/*test_ucp_am_id.am_id_overflow* ./test/gtest/gtest

[     INFO ] ugni is not available
[     INFO ] ugni,cuda_copy,rocm_copy is not available
[     INFO ] Using random seed of 10211
Note: Google Test filter = *rc/*test_ucp_am_id.am_id_overflow*
[==========] Running 2 tests from 1 test suite.
[----------] Global test environment set-up.
[----------] 2 tests from rc/test_ucp_am_id
[ RUN      ] rc/test_ucp_am_id.am_id_overflow/0 <rc_v>
invalid AM id 4294901761, must be in range [0, 65535]
invalid AM id 4294901761, must be in range [0, 65535]
[1719478243.466888] [rock28:19096:0]        spinlock.c:29   UCX  WARN  ucs_recursive_spinlock_destroy() failed: busy
common/test.cc:366: Failure
Failed
Got 0 errors and 1 warnings during the test
[     INFO ] < type/spinlock.c:29 ucs_recursive_spinlock_destroy() failed: busy >
[  FAILED  ] rc/test_ucp_am_id.am_id_overflow/0, where GetParam() = rc_v (347 ms)
[ RUN      ] rc/test_ucp_am_id.am_id_overflow/1 <rc_v/proto_v1>
invalid AM id 4294901761, must be in range [0, 65535]
invalid AM id 4294901761, must be in range [0, 65535]
[1719478243.741695] [rock28:19096:0]        spinlock.c:29   UCX  WARN  ucs_recursive_spinlock_destroy() failed: busy
common/test.cc:366: Failure
Failed
Got 0 errors and 1 warnings during the test
[     INFO ] < type/spinlock.c:29 ucs_recursive_spinlock_destroy() failed: busy >
[  FAILED  ] rc/test_ucp_am_id.am_id_overflow/1, where GetParam() = rc_v/proto_v1 (274 ms)
[----------] 2 tests from rc/test_ucp_am_id (621 ms total)

[----------] Global test environment tear-down
[==========] 2 tests from 1 test suite ran. (623 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 2 tests, listed below:
[  FAILED  ] rc/test_ucp_am_id.am_id_overflow/0, where GetParam() = rc_v
[  FAILED  ] rc/test_ucp_am_id.am_id_overflow/1, where GetParam() = rc_v/proto_v1

 2 FAILED TESTS
```

Have you seen something like this?

Unless it's a silly mistake and I missed something, please let me know if you want me to keep searching for the cause in the scope of this PR.